lib/internal/child_process.js

```diff
@@ -526,6 +533,37 @@ ChildProcess.prototype.unref = function() {
   if (this._handle) this._handle.unref();
 };
 
+ChildProcess.prototype.readLines = function readLines(options = undefined) {
+  const { useStdErr = false, ignoreErrors = false } = options ?? kEmptyObject;
+  if (options != null) {
+    validateBoolean(useStdErr, 'options.useStdErr');
+    validateBoolean(ignoreErrors, 'options.ignoreErrors');
+  }
+  const rlInterface = new Interface({
+    __proto__: null,
+    signal: this[kSignal],
+    input: useStdErr ? this.stderr : this.stdout,
+  });
+  if (ignoreErrors) {
+    this.on('error', () => {});
+  } else {
+    const errorPromise = new Promise((_, reject) => this.once('error', reject));
+    const exitPromise = new Promise((resolve, reject) => this.once('exit', (code) => {
```

Comment:
You would need to listen for `close`, not `exit` here, otherwise you’d run the risk of dropping data.